### ISSUE TITLE
Restoring from backups returns error due to missing database password

### ISSUE THREAD

**Issue by zack-hable** [litespeedtech/cyberpanel#260]
I've recently had to transfer some sites to a new machine and was experiencing errors in the web UI when I would restore a website.  This only seemed to happen to some of my sites and I narrowed down the issue to be that the initial backups had a partially formed <password> field in the meta.xml file and would result in it being None when later read to be restored.  
I'm not sure why the database passwords aren't being backed up in the meta.xml file, but it may be nice to check values prior to usage in plogical/mysqlUtilities.py (line 772) and let the user know.
Excerpt of meta.xml:
```
<?xml version="1.0" ?>
<metaFile>
  <masterDomain>MY_SITE.COM</masterDomain>
  <phpSelection>PHP 7.3</phpSelection>
  <externalApp>MY_APP</externalApp>
  <ChildDomains/>
  <Databases>
    <database>
      <dbName>MY_DB_NAME</dbName>
      <dbUser>MY_DB_USER</dbUser>
      <password/>
    </database>
  </Databases>
```
OS: Ubuntu 18.04
CyberPanel: 1.9.4

**Comment by usmannasir** [comment]
These sites were restored from cPanel?

**Comment by zack-hable** [comment]
I imported these manually from Webmin/Virtualmin.  So I made all of the email accounts and databases manually through the cyberpanel admin.

**Comment by usmannasir** [comment]
in `/usr/local/CyberCP/CyberCP/settings.py` there is an entry which requires MySQL root password, can you confirm if that password is correct?

**Comment by zack-hable** [comment]
Yes, as I said earlier some sites were successfully restored while others (example above) weren't.  For reference I'm able to import all of my failed sites by changing:
```
if passwordCheck == None:
```
to
```
if passwordCheck == None and dbPassword is not None:
```
on line [238](https://github.com/usmannasir/cyberpanel/blob/b5c0c169fd97a64036c077be9435421b091e2bdc/plogical/mysqlUtilities.py#L238)

**Comment by usmannasir** [comment]
can you confirm that changing that line skipped the malfunction part or sites restored properly?

**Comment by zack-hable** [comment]
I can confirm that it has imported my DB with all of its previous data and all I had to do was then add the password back to the user manually (I don't really see a way around this since the meta.xml file is missing the password field, but it would be nice to avoid digging around to find the issue.  I'm not sure if this is due to me possibly setting up my sites wrong initially)
The original log output only contained:
```
[02.15.2020_00-38-05] We had errors while creating DKIM record for: MY_DOMAIN. Error message: name 'ProcessUtiities' is not defined
[02.15.2020_00-38-10] Trying to obtain SSL for: mail.MY_DOMAIN and: www.mail.MY_DOMAIN
[02.15.2020_00-38-10] /root/.acme.sh/acme.sh --issue -d mail.MY_DOMAIN -d www.mail.MY_DOMAIN --cert-file /etc/letsencryp
t/live/mail.MY_DOMAIN/cert.pem --key-file /etc/letsencrypt/live/mail.MY_DOMAIN/privkey.pem --fullchain-file /etc/letsenc
rypt/live/mail.MY_DOMAIN/fullchain.pem -w /home/MY_DOMAIN/public_html/mail.MY_DOMAIN --force
[02.15.2020_00-38-19] Failed to obtain SSL for: mail.MY_DOMAIN and: www.mail.MY_DOMAIN
[02.15.2020_00-38-19] Trying to obtain SSL for: mail.MY_DOMAIN
[02.15.2020_00-38-26] Failed to obtain SSL, issuing self-signed SSL for: mail.MY_DOMAIN
[02.15.2020_00-38-26] Self signed SSL issued for mail.MY_DOMAIN.
[02.15.2020_00-38-31] must be str, not NoneType[restoreDatabaseBackup]
[02.15.2020_00-38-31]  [startRestore]
```

The edit I made only checks for the password field being present as if its None when you concatenate to set the mysql user's password an exception will be thrown.  I guess alternatively you could change the concatenation to use str(dbPassword) instead of dbPassword?

**Comment by usmannasir** [comment]
thanks @zack-hable 

Good find, this seems a sneaky bug. I need to find out why DB password is none. However, thanks for pin-pointing.

It is better you can join us on slack -> https://join.slack.com/t/cyberpanel/shared_invite/enQtNDc3MDA5NjM0MTMyLTY4YTQ3Y2JmYmFjODE4MGQ4YTI4NTQ1ZjFlMTkzZmNmOTY0YWU1NmVlNjZiNmFkNWU0ZjkzOWVkYmFiMDBkOWQ

Thanks again.

**Comment by master3395** [comment]
Thanks for the report. This issue is older and may not apply to current releases.

Please confirm:
1) CyberPanel version + build (please retest on the latest public release **2.4.9**)
2) OS name and version (whatever you run, e.g. `cat /etc/os-release`)
3) Exact steps still failing on a current install
4) Relevant logs (no passwords)

Note: the latest public release is **2.4.9**. Development branches are not recommended for reproducing bugs.

If we do not hear back within 30 days, this issue may be closed as stale. You can reopen with fresh details anytime.
